Template.php

```diff
@@ -2916,6 +2916,12 @@ public function tidy_parameter($param) {
         case 'url':
           if (preg_match("~^https?://(?:www\.|)researchgate\.net/[^\s]*publication/([0-9]+)_*~i", $this->get($param), $matches)) {
               $this->set($param, 'https://www.researchgate.net/publication/' . $matches[1]);
+              if (preg_match('~^(.+){{!}} Request PDF$~', trim($this->get('title')), $match)) {
```

Comment:
Do `{}`s need escaping?
```suggestion
              if (preg_match('~^(.+)\{\{!\}\} Request PDF$~', trim($this->get('title')), $match)) {
```

(Is it worth including a test that would catch this?)